domain/state.go

```diff
@@ -121,15 +128,14 @@ func (st *StateBase) RunAtomic(ctx context.Context, fn func(AtomicContext) error
 		// transaction from being used outside of the transaction scope. This
 		// will prevent any references to the sqlair.TX from being held outside
 		// of the transaction scope.
+		txCtx := st.atomicPool.Get().(*atomicContext)
+		txCtx.ctx = ctx
+		txCtx.tx = tx
 
-		// TODO (stickupkid): The atomicContext can be pooled on the StateBase
-		// to reduce the number of allocations. Attempting to push the tx into
-		// the context would prevent that as a viable option.
-		txCtx := &atomicContext{
-			ctx: ctx,
-			tx:  tx,
-		}
-		defer txCtx.close()
+		defer func() {
+			txCtx.close()
+			st.atomicPool.Put(txCtx)
```

Comment:
Shouldn't we add the transaction to the pool right after we assign `tx` on it (line 133)? Why on defer?

Reply:
We get it from the pool on line 131 and then we need to put it back in the pool on line 137 when we're done with the atomic context. Otherwise, it'll just keep creating them.

Reply:
Oh yes right, it has to be done at the same time of txCtx close. Thanks.